relay-sampling/src/lib.rs

```diff
@@ -3219,6 +3281,80 @@ mod tests {
         serde_json::from_str::<DynamicSamplingContext>(json).unwrap_err();
     }
 
+    #[test]
+    fn test_parse_sampled_with_incoming_boolean() {
+        let json = r#"
+        {
+            "trace_id": "00000000-0000-0000-0000-000000000000",
+            "public_key": "abd0f232775f45feab79864e580d160b",
+            "user_id": "hello",
+            "sampled": true
+        }
+        "#;
+        let dsc = serde_json::from_str::<DynamicSamplingContext>(json).unwrap();
+        insta::assert_ron_snapshot!(dsc, @r###"
```

Comment:
Instead of RON, could you assert JSON here? This should roundtrip cleanly, since we require the upstream to parse this again.